internal/ceb/config.go

```diff
@@ -77,13 +61,76 @@ func (ceb *CEB) initConfigStream(ctx context.Context, cfg *config, isRetry bool)
 
 // watchConfig sits in a goroutine receiving the new configurations from the
 // server.
-func (ceb *CEB) watchConfig(ch <-chan *pb.EntrypointConfig) {
+func (ceb *CEB) watchConfig(
+	log hclog.Logger,
+	ch <-chan *pb.EntrypointConfig,
+	ctx context.Context,
+	cfg *config,
+) {
+	// Keep track of our currently executing command information so that
+	// we can diff properly to determine if we need to restart.
+	currentCmd := ceb.copyCmd(ceb.childCmdBase)
+
+	// We only init the URL service once. In the future, we can do diffing
+	// and support automatically reinitializing if the URL service changes.
+	didInitURL := false
+
 	for config := range ch {
+		if !didInitURL {
```

Comment:
I guess we don't expect the server to change the token or or address or anything for the UrlService?

Reply:
Yeah for now. Note the comment when this variable is defined. This is just the same behavior as prior to this PR. We can add support for changing the URL service config later!

Reply:
No problem! Just wanted to double check the assumption.